### PR TITLE
RF-916: Fix initial workflow for PII screen

### PR DESCRIPTION
**Pull request by KramerJeff**
JIRA: https://gathering.atlassian.net/browse/RF-916

Fix initial workflow on PII screen so date of birth has proper
validation. Change masking package to formatStringByPattern due to
backspaces not being read if user entered more characters than needed
in a specific pattern.

QA - Check PII screen when setting up a new customer and ensure the phone number, date of birth, and social security number inputs are properly validating user input

### CODE REVIEW

**Review by ksalmon — CHANGES_REQUESTED** [review]
Go back and read the ticket's second requirement. We need to do a better job at handling the error if the PII PUT is not a proper request. For example if the SSN already exists in the program you will get an error

File: screens/PIIScreen.tsx [hunk 1]
```diff
@@ -89,7 +90,8 @@ export default function PIIScreen({ navigation }: PIIScreenProps): JSX.Element {
     last_name: Yup.string().required('Last Name is required.'),
     dob: Yup.date()
       .required('Date of Birth is required.')
-      .max(maxDob, 'You should be at least 18 years old.'),
+      .max(maxDob, 'You should be at least 18 years old.')
```
Comment (RockinRonE) [inline screens/PIIScreen.tsx:93]: I'm able to add letters in the DOB field...while I get a validation error, should we prevent the user from doing so?

File: screens/PIIScreen.tsx [hunk 2]
```diff
@@ -228,15 +233,13 @@ export default function PIIScreen({ navigation }: PIIScreenProps): JSX.Element {
                 errorText={!touched.city ? '' : errors.city}
                 editable={!isSubmitting}
               />
-              <Input
-                placeholder="State"
-                onChangeText={handleChange('state')}
-                onBlur={handleBlur('state')}
+              <Dropdown
```
Comment (RockinRonE) [inline screens/PIIScreen.tsx:236]: Let's have the background color, border, etc. to match the other inputs.

**Review by RockinRonE — CHANGES_REQUESTED** [review]

**Review by ksalmon — APPROVED** [review]

File: screens/ConfirmPIIScreen.tsx [hunk 3]
```diff
@@ -74,7 +74,12 @@ export default function ConfirmPIIScreen({
           navigation.navigate('BankingDisclosures');
         },
         (error) => {
-          setErrorText(error.message);
+          if (error[0]) {
```
Comment (ksalmon) [inline screens/ConfirmPIIScreen.tsx:77]: What happens if there is more than one error? I would prefer it if we could concatenate and display them all.

**Review by ksalmon — CHANGES_REQUESTED** [review]

File: screens/ConfirmPIIScreen.tsx [hunk 4]
```diff
@@ -74,7 +74,12 @@ export default function ConfirmPIIScreen({
           navigation.navigate('BankingDisclosures');
         },
         (error) => {
```
Comment (ksalmon) [inline screens/ConfirmPIIScreen.tsx:76]: This makes sense to call it `errors`

**Review by ksalmon — CHANGES_REQUESTED** [review]

**Review by ksalmon — APPROVED** [review]
LGTM

File: screens/PIIScreen.tsx (hunk 2 above)
Comment (RockinRonE) [reply screens/PIIScreen.tsx:236]: @ksalmon @KramerJeff I found if you move the input styles of the DropDown from `input` to `container` it at least styles the input.

Specifically moving: 
`
borderRadius: 4,
      borderColor: border,
      borderWidth: 2,
`
from input to container

🤷 about the placeholder

Up to you all how far you want to go, but this seemed easy enough

**Review by RockinRonE — APPROVED** [review]
Not gonna hold this up about styling, but I left a comment about some touchup that can be done

**Pull request merged**